validator/length.go

```diff
@@ -0,0 +1,127 @@
+package validator
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/spf13/cobra"
+)
+
+// Length is a struct that provides a map
+// with key as attribute for which length is controlled
+// and value limit as Limit struct
+type Length struct {
+	Limits map[string]Limit
+}
+
+// Limit defines min, max length of string
+type Limit struct {
+	Min, Max int
+}
+
+// Validate implements the Rule interface
+func (l *Length) Validate(cmd *cobra.Command, verbose bool) []Error {
+	var errors []Error
+	info := stats{num: 0, num_failed: 0, errors: errors}
+
+	return l.ValidateHelper(cmd, verbose, info)
+}
+
+// ValidateHelper returns errors of type []Error
+// if length is not validated
+func (l *Length) ValidateHelper(cmd *cobra.Command, verbose bool, info stats) []Error {
+
+	// validate the root command
+	err := validateLength(cmd, l, verbose)
+	// record stats
+	info.num++
+	info.num_failed += len(err)
+	info.errors = append(info.errors, err...)
+
+	// traverse descendents of cmd
+	for _, child := range cmd.Commands() {
+
+		// base case
+		if !child.IsAvailableCommand() || child.IsAdditionalHelpTopicCommand() {
+			continue
+		}
+
+		// recursive call for ValidateHelper
+		if err := l.ValidateHelper(child, verbose, info); err != nil {
+			return err
+		}
+	}
+
+	// prints additional info in debug mode
+	if verbose {
+		fmt.Printf("commands checked: %d\nchecks failed: %d\n", info.num, info.num_failed)
+	}
+
+	return info.errors
+}
+
+// ValidateLength traverse over all the fields
+// provided and compare with the limit
+func validateLength(cmd *cobra.Command, l *Length, verbose bool) []Error {
+	var errors []Error
+
+	for fieldName, limits := range l.Limits {
+		// reflects the fieldName in cobra.Command struct
+		reflectValue := reflect.ValueOf(cmd).Elem().FieldByName(fieldName)
+
+		// if the defined fieldName doesn't exist in cobra.Command
+		if reflectValue.String() == "<invalid Value>" {
```

Comment:
I think that reflect has method for checking this before we call String

Reply:
that was giving panic: reflect: call of reflect.Value.Type on zero Value